python/sidecar/references/service.py

```diff
@@ -12,9 +14,51 @@
 from sidecar.shared import chunk_reference
 
 
+def is_fetchable_pdf(url: str) -> Tuple[bool, str]:
+    """
+    Determines if a URL points to a PDF that can be fetched.
+
+    Parameters
+    ----------
+    url : str
+        The URL of the PDF to check.
+
+    Returns
+    -------
+    bool
+        Whether the URL points to a PDF that can be fetched.
+    str
+        The reason why the PDF could not be fetched.
+    """
+    try:
+        response = requests.get(url)
+
+        if not response.ok:
+            reason = f"Unable to fetch {url}, status code {response.status_code}"
+            logger.info(reason)
+            return False, reason
+
+        if response.status_code == 302:
+            location = response.headers.get("location")
+
+            if location:
+                return is_fetchable_pdf(location)
+
+        if "application/pdf" not in response.headers.get("content-type", ""):
+            reason = f"Unable to fetch {url}, content-type is not PDF"
+            logger.info(reason)
+            return False, reason
+
+        return True, ""
+    except requests.exceptions.RequestException:
+        reason = f"Unable to fetch {url}, request exception"
+        logger.info(reason)
+        return False, reason
```

Comment:
@jvoytek I added some reasons here that we pass back to the user, but I don't really love them. Maybe we have something more user-friendly on the frontend that suggests they visit the URL and manually add the paper, since that's ultimately what all of these are going to boil down to.

Also happy to change them here if you have suggestions.

Reply:
Added a more verbose tooltip when response contains a message